### ISSUE TITLE
"Unable to deserialize Real Debrid API response to Dictionary"

### ISSUE THREAD

**Issue by jeradsloan** [rogerfar/rdt-client#125]
hey! loving rdt client. thanks for the awesome work.
started getting this error on occasional files recently.  Feel free to mark this as closed, just wanted to share this in case it was useful

Unable to deserialize Real Debrid API response to Dictionary`2. Response was: { "9f0c05478e3d25f8905206de48e182e5c291d6f9": [ ] } 

---
RDT Client Version 2.0.9
running in Docker on ARM64
using Real-Debrid

<img width="1024" alt="Screen Shot 2022-03-14 at 3 00 10 PM" src="https://user-images.githubusercontent.com/13736/158268659-021f51a3-3ac1-4bf5-a863-b51e89438ea6.png">

**Comment by rogerfar** [comment]
If you click on "Torrent Files" what does it show for selected files?

**Comment by kloe7** [comment]
I have this same issue when more than one file gets sent to rdt-client from sonarr.  The first file downloads fine then I get this error for the remaining episodes. I have to delete the torrent files and restart the Docker container. Will continue to work fine until more than one episode gets sent to rdt-client. 

To answer your question it shows a red x beside every file listed in selected files

**Comment by jeradsloan** [comment]
here's what it shows.
<img width="1022" alt="Screen Shot 2022-03-15 at 9 01 45 PM" src="https://user-images.githubusercontent.com/13736/158514442-cf37d074-7990-4266-b705-7481b4098c3c.png">

anecdotally... I got the same results once when manually adding a torrent by clicking on ADD NEW TORRENT. gave same error and would not contineu.  But when I changed "Download action" to "Download all available files" (from the default "Download all files") it continued without error.

**Comment by kloe7** [comment]
I went into rdt-client settings -> Radarr Sonarr -> Set a minimum file limit and unchecked Only download available files on Real-Debrid. Everything seems to be running fine now for me.

**Comment by 0Blake** [comment]
I'm suddenly seeing this issue now after updating to 2.0.9. I do have the minimum file size set to 10mb and the "Only download available files on Real-Debrid" option unchecked.

**Comment by rogerfar** [comment]
Something is up with Real Debrid for sure, for example the torrent with hash `8EB17E22D70287B7013115A191305913CB2B2165`. When I add it to Real-Debrid, either manually or through the API, I get this popup on Real-Debrid:

![image](https://user-images.githubusercontent.com/1142915/159131127-f057978b-575b-4058-82a0-2e825e96d0e7.png)
Indicating that the top file is available for download immediatly.

But when I query with the API I get this result:
```
{
	"8eb17e22d70287b7013115a191305913cb2b2165": [
		
	]
}
```
Showing no files available.

This is odd, because that's not even a valid result, when there are no available files the API is supposed to return:
```
{
	"dd8255ecdc7ca55fb0bbf81323d87062db1f6d1c": {
		"rd": [
			
		]
	}
}
```

**Comment by a597232887** [comment]
"settings -> Radarr Sonarr -> Set a minimum file limit and unchecked Only download available files on Real-Debrid" seems working.

**Comment by jinetjose** [comment]
This is what i get when trying to add a new torrent using magnet link

Please help issue started with 2.09 , earlier everything was working flawlessly 


---> Newtonsoft.Json.JsonSerializationException: Cannot deserialize the current JSON array (e.g. [1,2,3]) into type 'System.Collections.Generic.Dictionary`2[System.String,System.Collections.Generic.List`1[System.Collections.Generic.Dictionary`2[System.String,RDNET.TorrentInstantAvailabilityFile]]]' because the type requires a JSON object (e.g. {"name":"value"}) to deserialize correctly.
To fix this error either change the JSON to a JSON object (e.g. {"name":"value"}) or change the deserialized type to an array or a type that implements a collection interface (e.g. ICollection, IList) like List<T> that can be deserialized from a JSON array. JsonArrayAttribute can also be added to the type to force it to deserialize from a JSON array.
Path '374bb8eaea212e6130e2918c91d61583a4844471', line 2, position 46.

**Comment by jinetjose** [comment]
this looks like a RD issue

**Comment by rogerfar** [comment]
Could anyone who sees this error get the actual JSON part out of the logs and post it here?

I have implemented a fix, but I don't know the JSON structure of the actual response, I think that the RD API is just broken for some torrents, but if it gives actual info back I would like to make sure it gets parsed correctly.

**Comment by rogerfar** [comment]
This is fixed in 2.0.10 now, if you still have issues please feel free to re-open this issue.
